PROTO_tests/tests/groupby_test.py

```diff
@@ -89,33 +98,75 @@ def test_pandas_equivalency(self, size, levels):
-                    do_check = False
-                try:
-                    akkeys, akvals = akg.aggregate(akdf[vname], op)
-                except Exception as E:
-                    do_check = False
-                    continue
-                if not do_check:
-                    continue
+            if op == "count":
+                print(f"Doing .count() - {vname}")
+            else:
+                print(f"\nDoing aggregate({vname}, {op})")
+
+            do_check = True
+            try:
+                pdkeys, pdvals = self.groupby_to_arrays(df, keyname, vname, op, levels)
+            except Exception:
+                print("Pandas does not implement")
+                do_check = False
+            try:
+                akkeys, akvals = akg.count() if op == "count" else akg.aggregate(akdf[vname], op)
+            except Exception as E:
+                print("Arkouda error: ", E)
+                continue # skip check
+            if do_check:
                 if op.startswith("arg"):
                     pdextrema = df[vname][pdvals]
                     akextrema = akdf[vname][akvals].to_ndarray()
+                    # check so we can get meaningful output if needed
+                    if not np.allclose(pdextrema, akextrema):
+                        print("Different argmin/argmax: Arkouda failed to find an extremum")
+                        print("pd: ", pdextrema)
+                        print("ak: ", akextrema)
                     assert np.allclose(pdextrema, akextrema)
                 else:
                     if op != "unique":
                         self.compare_keys(pdkeys, akkeys, levels, pdvals, akvals)
 
+    @pytest.mark.parametrize("size", pytest.prob_size)
+    @pytest.mark.parametrize("op", NAN_OPS)
+    def test_pandas_equivalency_nan(self, size, op):
+        d = self.make_arrays_nan(size)
+        df = pd.DataFrame(d)
+        akdf = {k: ak.array(v) for k, v in d.items()}
+
+        akg = ak.GroupBy(akdf["keys"])
+        keyname = "keys"
+
+        if op == "count":
+            print(f"Doing .count()")
+        else:
+            print(f"\nDoing aggregate(float64, {op})")
+
+        do_check = True
+        try:
+            pdkeys, pdvals = self.groupby_to_arrays(df, keyname, "float64", op, 1)
+        except Exception:
+            print("Pandas does not implement")
+            do_check = False
+        try:
+            akkeys, akvals = akg.count() if op == "count" else akg.aggregate(akdf["float64"], op, True)
+        except RuntimeError as E:
+            print("Arkouda error: ", E)
+            do_check = False
+        if do_check:
+            for i in range(pdvals.size):
+                if np.isnan(pdvals[i]):
+                    pdvals[i] = 0.0  # clear out any nans to match ak implementation
+            print()
+            print(akg.keys.to_list())
+            print(pdkeys)
+            print(pdvals)
+            print(akkeys)
+            print(akvals)
```

Comment:
these prints seem to be leftover debugging for the test but I might be wrong about that